src/rules/unbound-method.ts

```diff
@@ -73,9 +74,58 @@ export default createRule<Options, MessageIds>({
       return {};
     }
 
+    /**
+     * Checks if a MemberExpression is an argument to a `jest.mocked()` call.
+     * This handles cases like `jest.mocked(service.method)` where `service.method`
+     * should not be flagged as an unbound method.
+     */
+    const isArgumentToJestMocked = (
+      node: TSESTree.MemberExpression,
+    ): boolean => {
+      // Check if the immediate parent is a CallExpression
+      if (node.parent?.type !== AST_NODE_TYPES.CallExpression) {
```

Comment:
this is the opposite of the check we do later on, so ideally we can merge the two but that might be messy due to the optional chaining and require an `else` which I try to avoid so not too fussed to keep it like this for now